### ISSUE TITLE
Code Inspections should be aware of the Project's Compilation Arguments

### ISSUE THREAD

**Issue by ThunderFrame** [rubberduck-vba/Rubberduck#927]
Choose VBIDE's Tools..[Project Name] Properties... menu, and you can define multiple Conditional compiler Arguments, separated by semi-colon, eg.

```
MyProjectArg = 2 : MyOtherArg = 3
```

You can then use those arguments in Conditional Blocks:

```
#If MyProjectArg = 2 Then
  #If MyProjectArg = MyOtherArg Then
    Const Name = "Smith"
  #Else
    Const Name = "Jackson"
  #End If
#Else
  Const Name = "Jones"
#End If
```

But I don't know a way of determining the Project's Conditional Argument string?

**Comment by ckuhn203** [comment]
This is sticky. We've tackled it before, but the only way I've found to do it is by opening the dialog box. I haven't found a programmatic way to do it. If anyone can figure out a way, I'd love to implement.

**Comment by ThunderFrame** [comment]
The only other way, that I can think of, is to read the PROJECTCONSTANTS Record (see MS OVBA document) inside the VBProject binary from the hosting file, but that would only give you the constants as they stood at last save.

**Comment by bclothier** [comment]
With type information API, we now can obtain project's compilation arguments, which should be taken into account at the start of parsing.

**Comment by retailcoder** [comment]
The preprocessor pass, specifically. Not sure this has anything to do with inspections, but with this knowledge the preprocessor pass can now evaluate *all* precompiler directives, i.e. we get the complete actual "live code" into parse trees.

**Comment by WaynePhillipsEA** [comment]
closed by #4715
